info-compiler/src/test/java/com/google/sps/NewsContentExtractorTest.java

```diff
@@ -77,33 +87,51 @@ public void extractContentFromHtml_realValidWebpageWithWrongUrlParam() {
     // URL. Other information should be consistent with {@code URL} and {@code NEWS_ARTICLE}.
     // Content processing hasn't occurred so the abbreivated content is null.
     Optional<NewsArticle> potentialNewsArticle =
-        NewsContentExtractor.extractContentFromHtml(realWebpageStream, WRONG_URL);
+        realNewsContentExtractor.extractContentFromHtml(realWebpageStream,
+                                                        NEWS_ARTICLE_WITH_WRONG_URL.getUrl());
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    NewsArticle newsArticle = potentialNewsArticle.get();
+    Assert.assertEquals(newsArticle, NEWS_ARTICLE_WITH_WRONG_URL);
+  }
+
+  @Test
+  public void extractContentFromHtml_emptyDocParser() throws IOException, SAXException,
+      TikaException {
+    // Extract content and meta data using a parser that modifies the content handler to
+    // receive an end-of-document notification via {@code endDocument()}. Therefore, an empty
+    // article will be extracted, while the URL is set correctly to the passed in parameter.
+    // Content processing hasn't occurred so the abbreivated content is null.
+    HtmlParser parser = mock(HtmlParser.class);
+    doAnswer(parseInvocation -> {
+               BoilerpipeContentHandler boilerpipeHandler =
+                   (BoilerpipeContentHandler) parseInvocation.getArguments()[1];
+               boilerpipeHandler.endDocument();
+               return null;
+            }).when(parser).parse(anyObject(), anyObject(), anyObject());
```

Comment:
And write more tests that do this, where you set up the boilerpipehandler to contain the TextDocument for the test case.